.gitlab-ci.yml

```diff
@@ -0,0 +1,146 @@
+#   artifacts:
+#     paths:
+#       - coverage/
+
+# test_js_job:
+#   stage: test
+#   variables:
+#     RAILS_ENV: development
+#     NODE_ENV: development
+#   before_script:
+#     - docker login -u gitlab-ci-token -p $CI_JOB_TOKEN $CI_REGISTRY
+#     - sed -i "s/vsv/\${RAILS_IMAGE}:\${CI_COMMIT_SHA}-test/" docker-compose.yml
+#     - sed -ie '/build:/,/precompileassets/d' docker-compose.yml
+#     - echo "PROJECT_NAME=${PROJECT_NAME}" >> .env
+#   # use docker-compose to prevent configuration duplication
+#   script:
+#     - docker-compose run web yarn test
+
+build_web_job:
+  stage:
+    build
+  before_script:
+    - touch .env
+    #- export CI_COMMIT_REF_NAME_UP=`echo $CI_COMMIT_REF_NAME | awk '{ print toupper($0) }' | sed 's/-/_/g'`
+    - echo "PROJECT_NAME=${PROJECT_NAME}" > .env
+    #- env | grep K8S_SECRET_
+    #- env | grep K8S_SECRET_ | sed "s/^K8S_SECRET_\(.*\)$/\1/g" | sed "s/^${CI_COMMIT_REF_NAME_UP}_\(.*\)$/\1/g" > .env
+    # - env
+    # - cat .env
+    - docker login -u gitlab-ci-token -p $CI_JOB_TOKEN $CI_REGISTRY
+  variables:
+    RAILS_ENV: production
+    NODE_ENV: production
+  script:
+    - echo "DATABASE_URL=${K8S_SECRET_DEVELOP_DATABASE_URL}" >> .env
+    # - cat .env
+    - docker build . -t ${RAILS_IMAGE}:${CI_COMMIT_SHA}
+    - docker push $RAILS_IMAGE
+  only:
+    refs:
+      - master
+      - production
+      - develop
+
+deploy_web_job:
+  stage: deploy
+  image:
+    name: lachlanevenson/k8s-kubectl:latest
+    entrypoint: ["/bin/sh", "-c"]
+  script:
+    - echo "Deploying to namespace ${KUBE_NAMESPACE}"
+    - create_application_secret
+    - kubectl version
+    - cd manifests/
+    - export CI_COMMIT_REF_NAME_UP=`echo $CI_COMMIT_REF_NAME | awk '{ print toupper($0) }'`
+    - echo $CI_COMMIT_REF_NAME_UP
+    - sed -i "s/__CI_COMMIT_SHA__/${CI_COMMIT_SHA}/" deployment.yaml ingress_${CI_COMMIT_REF_NAME_UP}.yaml service.yaml worker.yaml
+    - sed -i "s/__CI_COMMIT_REF_NAME__/${CI_COMMIT_REF_NAME_UP}/" deployment.yaml ingress_${CI_COMMIT_REF_NAME_UP}.yaml service.yaml worker.yaml
+    - kubectl apply -f deployment.yaml ingress_${CI_COMMIT_REF_NAME_UP}.yaml service.yaml worker.yaml
+    - kubectl rollout status -f deployment.yaml
+    - kubectl get all,ing -l app=${CI_ENVIRONMENT_SLUG}
+  only:
+    refs:
+      - master
+      - develop
+
+
+.deploy_helpers: &deploy_helpers |
+  # Extracts variables prefixed with K8S_SECRET_
+  # and creates a Kubernetes secret.
+  #
+  # e.g. If we have the following environment variables:
+  #   K8S_SECRET_A=value1
+  #   K8S_SECRET_B=multi\ word\ value
+  #
+  # See: https://gitlab.com/gitlab-org/gitlab-ce/blob/master/lib/gitlab/ci/templates/Jobs/Deploy.gitlab-ci.yml
+  #
+  # Then we will create a secret with the following key-value pairs:
+  #   data:
+  #     A: dmFsdWUxCg==
+  #     B: bXVsdGkgd29yZCB2YWx1ZQo=
+  function create_application_secret() {
+    track="${1-stable}"
+    export APPLICATION_SECRET_NAME=production-questionaire-secret
```

Comment:
Before it is too late, let's please rename this to `production-question*n*aire-secret` (you can do a simple search/replace over all files in the repo because it occurs in the yaml files from the deployments also)